docs/docs/migrating-from-v1-to-v2.md

```diff
@@ -175,22 +179,203 @@ export default props => (
+    <Img resolutions={data.bar.childImageSharp.resolutions} />
+  </div>
+}
+
+export default Example
+
+export const pageQuery = graphql`
+  query IndexQuery {
+    foo: file(relativePath: { regex: "/foo.jpg/" }) {
+      childImageSharp {
+        sizes(maxWidth: 700) {
+          ...GatsbyImageSharpSizes_tracedSVG
+        }
+      }
+    }
+    bar: file(relativePath: { regex: "/bar.jpg/" }) {
+      childImageSharp {
+        resolutions(width: 500) {
+          ...GatsbyImageSharpResolutions_withWebp
+        }
+      }
+    }
+  }
+`
+```
+
+After:
+
+```jsx{3-4,14-15,21-22}
+const Example = ({ data }) => {
+  <div>
+    <Img fluid={data.foo.childImageSharp.fluid} />
+    <Img fixed={data.bar.childImageSharp.fixed} />
+  </div>
+}
+
+export default Example
+
+export const pageQuery = graphql`
+  query IndexQuery {
+    foo: file(relativePath: { regex: "/foo.jpg/" }) {
+      childImageSharp {
+        fluid(maxWidth: 700) {
+          ...GatsbyImageSharpFluid_tracedSVG
+        }
+      }
+    }
+    bar: file(relativePath: { regex: "/bar.jpg/" }) {
+      childImageSharp {
+        fixed(width: 500) {
+          ...GatsbyImageSharpFixed_withWebp
+        }
+      }
+    }
+  }
+`
+```
+
+Further examples can be found in the [Gatsby Image docs](https://github.com/gatsbyjs/gatsby/tree/d0e29272ed7b009dae18d35d41a45e700cdcab0d/packages/gatsby-image).
+
+## Manually specify postcss plugins
+
+Gatsby v2 removed `postcss-cssnext` and `postcss-import` from the default postcss setup.
+
+Use [`onCreateWebpackConfig`](/docs/add-custom-webpack-config) to specify your postcss plugins.
+
+Note: there will be a `postcss` plugin that allows you to configure postcss from a standard postcss config file. [Follow this discussion on issue 3284](https://github.com/gatsbyjs/gatsby/issues/3284).
+
+## Files mixing CommonJS with ES6 modules won't compile
+
+Gatsby v2 uses babel 7 which is stricter about parsing files with mixed JS styles.
+
+ES6 Modules are ok:
+
+```js
+import foo from "foo"
+export default foo
+```
+
+CommonJS is ok:
+
+```js
+const foo = require('foo');
+module.exports = foo;
+```
+
+Mixing `requires` and `export` is not ok:
+```js
+const foo = require('foo');
+export default foo
+```
+
+Mixing `import` and `module.exports` is not ok:
+```js
+import foo from "foo"
+module.exports = foo;
+```
+
+See [Gatsby's babel docs for more details](/docs/babel).
+
+## Don't query nodes by ID
+
+Source and transformer plugins now use UUIDs for IDs. If you used glob or regex to query nodes by id then you'll need to query something else.
+
+[See the Pull Request that implemented this change](https://github.com/gatsbyjs/gatsby/pull/3807/files)
+
+> TODO: add example
+
+## Remove explicit polyfills
+
+If your Gatsby v1 site included any polyfills, you can remove them. Gatsby v2 ships with babel 7 and is configured to automatically include polyfills for your code. See [Gatsby's babel docs for more details](/docs/babel).
```

Comment:
This doesn't appear to be actually true, or I'm misunderstanding. I had to ship an IntersectionObserver polyfill for marisamorby.com:

https://github.com/jlengstorf/marisamorby.com/pull/8/files#diff-3863fe9c963d5c58d29a7f5af77e6480

Reply:
Gatsby isn't yet compiling code in `node_modules` so code there won't be polyfilled yet. See bullet number 5 https://github.com/gatsbyjs/gatsby/issues/3870